src/main/java/refactoringml/RunImportCsv.java

```diff
@@ -0,0 +1,165 @@
+import java.util.stream.Stream;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.SessionFactory;
+
+import refactoringml.db.Database;
+import refactoringml.db.HibernateConfig;
+import refactoringml.db.HibernateConfig.ConnectionPoolType;
+import refactoringml.db.HibernateConfig.DatabaseEngine;
+import refactoringml.util.PropertiesUtils;
+
+public class RunImportCsv {
+	private static final Logger log = LogManager.getLogger(RunImportCsv.class);
+
+	public static void main(String[] args) throws Exception {
+		RunImportCsv runner = new RunImportCsv();
+		Args argv = new Args();
+		JCommander.newBuilder().addObject(argv).args(args).build();
+		cleanOldTempDir();
+		runner.run(argv, "/tmp/repos");
+	}
+
+	private static class Args {
+
+		@Parameter(names = { "-h", "--db-host" }, description = "Host IP or address of database")
+		private String databaseHost = "127.0.0.1";
+
+		@Parameter(names = { "-P", "--db-port" }, description = "Port of database")
+		private Short databasePort;
+
+		@Parameter(names = { "-u", "--db-user" }, description = "Database user")
+		private String databaseUser = System.getProperty("user.name");
+
+		@Parameter(names = { "-p", "--db-password" }, description = "Database password", password = true)
+		private String databasePassword = "pass";
+
+		@Parameter(names = { "-n", "--db-name" }, description = "Name of the database to use")
+		private String databaseName = "refactoringdb";
+
+		// TODO use converter into Enum directly instead of valueOf later on based on
+		// string to construct enum.
+		@Parameter(names = { "-d",
+				"--db-engine" }, description = "The database engine being used, currently supported values are MYSQL, MARIADB or POSTGRESQL")
+		private String databaseEngine = "MARIADB";
+
+		// TODO use converter into Enum directly instead of valueOf later on based on
+		// string to construct enum.
+		@Parameter(names = { "-c",
+				"--connection-pool" }, description = "The connection pooling library used. currently supported values are MYSQL, MARIADB or POSTGRESQL")
+		private String connectionPoolType = "HIKARICP";
+
+		@Parameter(names = { "-b", "--hbm2ddl" }, description = "The hibernate.hbm2ddl.auto in the database connection")
+		private String hbm2ddlStrategy = "create";
+
+		@Parameter(names = { "-s", "--store-source-code" }, description = "Store the source code of the metrics")
+		private boolean storeSourceCode = false;
+
+		@Parameter(names = { "-t",
+				"thread-amount" }, description = "Amount of threads to run in parallel. Will default to amount of cores available")
+		private int runWithCores = Runtime.getRuntime().availableProcessors();
+
+		@Parameter(required = true, description = "Input csv file")
+		private String inputCsvFile;
+
+	}
+
+	private void run(Args args, String repoLocation) throws InterruptedException, IOException {
```

Comment:
How does this behave under high load? E.g. huge projects with enormous files?

Reply:
Could you give me an example of a project with such files?. Then i can test it. I tested the first 1000 projects in projects_final.csv and there seemed to be no issues.